src/parser/SparqlParser.cpp

```diff
@@ -328,22 +245,11 @@ SparqlFilter SparqlParser::parseRegexFilter(bool expectKeyword) {
 template <typename ContextType>
 auto SparqlParser::parseWithAntlr(
     ContextType* (SparqlAutomaticParser::*F)(void),
-    const ParsedQuery& parsedQuery)
-    -> decltype((std::declval<sparqlParserHelpers::ParserAndVisitor>())
-                    .parseTypesafe(F)
-                    .resultOfParse_) {
-  return parseWithAntlr(F, getPrefixMap(parsedQuery));
-}
-
-// ________________________________________________________________________
-template <typename ContextType>
-auto SparqlParser::parseWithAntlr(
-    ContextType* (SparqlAutomaticParser::*F)(void),
-    SparqlQleverVisitor::PrefixMap prefixMap)
+    SparqlQleverVisitor::PrefixMap prefixMap, std::string query)
```

Comment:
Changed this to directly take the string to parse for two reasons:
1. The SPARQL parser is now only called from one place so there is no big overhead.
2. This allows to pass the original query  string (unmodified by the lexer which lowercases parts and inserts white space) to ANTLR code. With this the `ExceptionMetadata can now definitely be used in the frontend.